assets/js/stripe.js

```diff
@@ -525,10 +525,33 @@ jQuery( function( $ ) {
 			);
 
 			if ( $( 'form#add_payment_method' ).length ) {
-				$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
+				// Let's try to confirm the card details with a Setup Intent.
+				// 1. Create the setup intent
+				// 2. Confirm the setup intent
+				console.log( wc_stripe_params.setup_intent_client_secret );
+				console.log( response.source );
+				console.log( wc_stripe_params.customer_id );
+				stripe.confirmCardSetup( wc_stripe_params.setup_intent_client_secret, { payment_method: response.source.id } )
+					.then( function( result ) {
+						if ( result.error ) {
+							console.error( result.error );
+							return $( document.body ).trigger( 'stripeError', result );
+						}
+						const setupIntent = result.setupIntent;
+
+						if ( 'succeeded' !== setupIntent.status ) {
+							// TODO: We should probably do something smart here, like figuring out what specific action is needed?
+							console.error( setupIntent );
+							return $( document.body ).trigger( 'stripeError', result );
+						}
+
+						$( wc_stripe_form.form ).off( 'submit', wc_stripe_form.form.onSubmit );
+						wc_stripe_form.form.submit();
+					} )
```

Comment:
The goal here is to use the setup intent created when variables are loaded into the JS from PHP, and confirm it with the source that was just created in the frontend.

We should probably scratch this approach, and instead proxy a SetupIntent create+confirm request through the store. This will require us setting up an API on the store that would accept a source ID. The API would then need to get the Customer ID for the current user (or create a Stripe customer if none exist), attach the customer to the provided source, **and then** use this updated source to create and confirm the setup intent.

Reply:
> We should probably scratch this approach, and instead proxy a SetupIntent create+confirm request through the store. 

I agree, which is why I did it [here](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1398/files#diff-6bd4bc69841bdae2ebeafedd36523219eef8b3651b99c701951b2a397e38a1f3R143).